src/components/table-view/TableView.tsx

```diff
@@ -0,0 +1,499 @@
+              {toolbarActions}
+            </>
+          ) : undefined
+        }
+        pagination={
+          showPagination ? (
+            <Pagination
+              itemCount={totalCount}
+              page={page}
+              perPage={perPage}
+              onSetPage={(_e, newPage) => onPageChange(newPage)}
+              onPerPageSelect={(_e, newPerPage) => onPerPageChange(newPerPage)}
+              variant={position === 'bottom' ? 'bottom' : undefined}
+              isCompact={position === 'top'}
+            />
+          ) : undefined
+        }
+      />
+    );
+  };
+
+  // Render loading skeleton
+  const renderSkeleton = () => (
+    <Table aria-label={ariaLabel} variant={variant === 'compact' ? TableVariant.compact : undefined} ouiaId={ouiaId}>
+      <SkeletonTableHead
+        columns={[
+          ...(selectable ? [''] : []),
+          ...columns.map((col) => columnConfig[col as keyof typeof columnConfig]?.label || col),
+          ...(renderActions ? [''] : []),
+        ]}
+      />
+      <SkeletonTableBody rowsCount={perPage} columnsCount={columnCount} />
+    </Table>
+  );
+
+  // Default empty state for no data
+  const defaultEmptyStateNoData = (
+    <EmptyState>
+      <EmptyStateHeader titleText="No data available" headingLevel="h4" icon={<EmptyStateIcon icon={CubesIcon} />} />
+      <EmptyStateBody>There are no items to display. Create a new item to get started.</EmptyStateBody>
+    </EmptyState>
+  );
+
+  // Default empty state for no results (with filters active)
+  const defaultEmptyStateNoResults = (
+    <EmptyState>
+      <EmptyStateHeader titleText="No results found" headingLevel="h4" icon={<EmptyStateIcon icon={SearchIcon} />} />
+      <EmptyStateBody>No items match the filter criteria. Remove all filters or clear all filters to show results.</EmptyStateBody>
+      {clearAllFilters && (
+        <EmptyStateFooter>
+          <EmptyStateActions>
+            <Button variant="link" onClick={clearAllFilters}>
+              Clear all filters
+            </Button>
+          </EmptyStateActions>
+        </EmptyStateFooter>
+      )}
+    </EmptyState>
+  );
+
+  // Render empty state
+  const renderEmptyState = () => (
+    <Table aria-label={ariaLabel} variant={variant === 'compact' ? TableVariant.compact : undefined} ouiaId={ouiaId}>
+      <Thead>
+        <Tr>
+          {selectable && <Th screenReaderText="Select" />}
+          {columns.map((col) => (
+            <Th key={col}>{columnConfig[col as keyof typeof columnConfig]?.label || col}</Th>
+          ))}
+          {renderActions && <Th screenReaderText="Actions" />}
+        </Tr>
+      </Thead>
+      <Tbody>
+        <Tr>
+          <Td colSpan={columnCount}>
+            {hasActiveFilters ? emptyStateNoResults || defaultEmptyStateNoResults : emptyStateNoData || defaultEmptyStateNoData}
+          </Td>
+        </Tr>
+      </Tbody>
+    </Table>
+  );
+
+  // Render table with data
+  const renderTable = () => {
```

Comment:
I think we can create the JSX directly. It will be there most of the time. Abstracting it to a function/component does not help much in this case.